src/Orleans.CodeGeneration.Build/build/Microsoft.Orleans.OrleansCodeGenerator.Build.targets

```diff
@@ -1,31 +1,34 @@
 <Project TreatAsLocalProperty="CodeGenDirectory;IsCore;MSBuildIsCore;TargetIsCore;TaskAssembly;OutputFileName;CoreAssembly;FullAssembly;GeneratorAssembly">
-  
+
   <PropertyGroup Condition="'$(OrleansCodeGeneratorAssembly)' != ''">
     <!-- OrleansCodeGeneratorAssembly is used here to override the MSBuildIsCore value during Orleans.sln builds -->
     <MSBuildIsCore></MSBuildIsCore>
     <TargetIsCore></TargetIsCore>
+    <!-- For non-windows OS we force .Net Core  -->
+    <MSBuildIsCore Condition="'$(OS)' != 'Windows_NT'">true</MSBuildIsCore>
+    <TargetIsCore Condition="'$(OS)' != 'Windows_NT'">true</TargetIsCore>
```

Comment:
These default values seem to contradict what @ReubenBond intended on the lines immediately above them... Is it correct?

Reply:
In case of non-windows we've no options so we set them explicitly, so later checks see these as non-empty values.